flutter-idea/src/io/flutter/run/daemon/DevToolsService.java

```diff
@@ -115,6 +130,56 @@ private void startServer() {
     });
   }
 
+  private void setUpWithDart() {
+    final GeneralCommandLine command =
+      createCommand(DartSdk.getDartSdk(project).getHomePath(), "dart", ImmutableList.of("devtools", "--machine"));
+    try {
+      this.process = new MostlySilentColoredProcessHandler(command);
+      this.process.addProcessListener(new ProcessAdapter() {
+        @Override
+        public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+          final String text = event.getText().trim();
+
+          if (text.startsWith("{") && text.endsWith("}")) {
+            // {"event":"server.started","params":{"host":"127.0.0.1","port":9100}}
+
+            try {
+              final JsonElement element = JsonUtils.parseString(text);
+
+              // params.port
+              final JsonObject obj = element.getAsJsonObject();
```

Comment:
In case `dart devtools --machine` changes, you should check for `event:server.started` here.